docs/mosaicking.rst

```diff
@@ -0,0 +1,310 @@
+be specified either as a :class:`~astropy.wcs.WCS` or a
+:class:`~astropy.io.fits.Header` instance. If this is specified as a
+:class:`~astropy.wcs.WCS` instance, the ``shape_out`` argument to
+:func:`~reproject.reproject_interp` should also be specified, and be
+given the shape of the output image using the Numpy ``(ny, nx)`` convention
+(this is because :class:`~astropy.wcs.WCS`, unlike
+:class:`~astropy.io.fits.Header`, does not contain information about image
+size).
+
+Finally, the ``reproject_function`` should be used to specify which function to
+use to reproject individual tiles - this should be either
+:func:`~reproject.reproject_interp` or :func:`~reproject.reproject_exact` - with
+the latter being slower but more accurate. Keyword arguments for these functions
+(e.g. ``order`` for :func:`~reproject.reproject_interp`) can be passed as keyword
+arguments to :func:`~reproject.mosaicking.reproject_and_coadd`.
+
+The example above will return an array which is the mosaic itself, and a
+footprint, which shows how many input images contributed to each output pixel.
+We can take a look at the output:
+
+.. plot::
+   :context: reset
+   :nofigs:
+
+    from astropy.io import fits
+    from astropy.coordinates import SkyCoord
+    from pyvo.dal import imagesearch
+
+    pos = SkyCoord.from_name('M17')
+    table = imagesearch('https://irsa.ipac.caltech.edu/cgi-bin/2MASS/IM/nph-im_sia?type=at&ds=asky&',
+                       pos, size=0.25).to_table()
+    table = table[(table['band'] == b'K') & (table['format'] == b'image/fits')]
+    m17_hdus = [fits.open(row['download'].decode('ascii'))[0] for row in table]
+
+    from astropy.coordinates import SkyCoord
+    from reproject.mosaicking import find_optimal_celestial_wcs
+    coord = SkyCoord.from_name('M17')
+    wcs_out, shape_out = find_optimal_celestial_wcs(m17_hdus,
+                                             reference=coord)
+
+    from reproject import reproject_interp
+    from reproject.mosaicking import reproject_and_coadd
+    array, footprint = reproject_and_coadd(m17_hdus,
+                                           wcs_out, shape_out=shape_out,
+                                           reproject_function=reproject_interp)
+
+.. plot::
+   :include-source:
+   :align: center
+   :context:
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    plt.figure(figsize=(10, 8))
+    ax1 = plt.subplot(1, 2, 1)
+    im1 = ax1.imshow(array, origin='lower', vmin=600, vmax=800)
+    ax1.set_title('Mosaic')
+    ax2 = plt.subplot(1, 2, 2)
+    im2 = ax2.imshow(footprint, origin='lower')
+    ax2.set_title('Footprint')
+
+In some cases, including the above example, each tile that was used to compute
+the mosaic has an arbitrary offset due e.g. to different observing conditions.
+The :func:`~reproject.mosaicking.reproject_and_coadd` includes an option to
+match the backgrounds (assuming a constant additive offset in each image)::
```

Comment:
comment on the algorithm used for background matching and whether it is adjustable

Reply:
I've added a paragraph and mentioned that there are no exposed arguments for controlling this for now. The internal SGD function I wrote does have some options but we don't expose this at a higher level, and I'm not convinced we need to, but we can always re-assess in future once this is actually used in production?